### PR TITLE
BaSD-MS-W8-010-Psychologist-resource

### PR DESCRIPTION
**Pull request by ortizjoaquin**
- Create Psychologist ✅
- Update Psychologist ✅
- Remove Psychologist ✅
- List Psychologists ✅

### CODE REVIEW

File: controllers/psychologists.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  });
+}
+
+// Get All Psychologysts
+const getPsychologists = (req, res) => {
+	res.status(200).json(Psychologists);
+}
+
+// Get One Psychologysts
+const getOnePsychologist = (req, res) => {
```
Comment (nicolasviola) [inline controllers/psychologists.js:23]: Fix indentation in this function. It has 4 space per tab and should be 2

File: controllers/psychologists.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
+// Variables
```
Comment (nicolasviola) [inline controllers/psychologists.js:1]: Delete comment

File: controllers/psychologists.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+const fs = require('fs');
+const Psychologists = require('../data/psychologists.json');
+
+// Code shortened
```
Comment (nicolasviola) [inline controllers/psychologists.js:5]: Delete comment

File: controllers/psychologists.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+const getPsychologistIndex = (req) => {
+  return (Psychologists.findIndex((item) => item.id === req.query.id));
+}
+const writeFile = (file, obj) => {
```
Comment (nicolasviola) [inline controllers/psychologists.js:9]: Add a white space here to separate the functions

File: controllers/psychologists.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	const psychologistIndex = getPsychologistIndex(req);
+	if (psychologistIndex != -1) {
+		res.status(200).json(Psychologists[psychologistIndex]);
+	}
+	else {
+		res.status(400).json({ msg: `No psychologist with the Id of ${req.query.id}` });
+	};
+}
+
+// Create New Psychologyst
```
Comment (nicolasviola) [inline controllers/psychologists.js:33]: Delete comment

File: controllers/psychologists.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    return res.status(400).send({
+      msg: 'The email is missing'
+    });
+  }
+  else if (!req.query?.password){
+    return res.status(400).send({
+      msg: 'The password is missing'
+    });
+  }
+
```
Comment (nicolasviola) [inline controllers/psychologists.js:65]: Delete space

File: controllers/psychologists.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+  const {
+    first_name,
+    last_name,
+    user_name,
+    email,
+    password,
+  } = req.query;
+
```
Comment (nicolasviola) [inline controllers/psychologists.js:73]: Delete space

File: controllers/psychologists.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  const newPsychologist = {
+    id: (parseInt((Psychologists.length) + 1)).toString(),
+    first_name,
+    last_name,
+    user_name,
+    email,
+    password,
+  }
+
```
Comment (nicolasviola) [inline controllers/psychologists.js:82]: Delete space

File: controllers/psychologists.js [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+  writeFile('./data/psychologists.json', Psychologists)
+  return res.status(201).send({
+    msg: 'Psychologist Created'
+  });
+}
+
+// Edit Existing Psychologist
+const editPsychologist = (req, res) => {
+  const psychologistIndex = getPsychologistIndex(req);
+  if (psychologistIndex != -1) {
```
Comment (nicolasviola) [inline controllers/psychologists.js:93]: Please change the comparison. Use !== instead of !=

File: controllers/psychologists.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+      return res.status(201).send({
+      msg: 'Psychologist Edited'
+    });
+  } else {
+    res.status(404).send('Psychologist not found');
+  };
+}
+
+// Delete Existing Psychologist
+const deletePsychologist = (req, res) => {
```
Comment (nicolasviola) [inline controllers/psychologists.js:112]: Fix indentation in this function

File: controllers/psychologists.js [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+  const {
+    first_name,
+    last_name,
+    user_name,
+    email,
+    password,
+  } = req.query;
+
+  const newPsychologist = {
+    id: (parseInt((Psychologists.length) + 1)).toString(),
```
Comment (nicolasviola) [inline controllers/psychologists.js:75]: This id could be duplicate in the future if we delete items at the beginning of the array. Please use this function to assign new ids: parseInt(Psychologists[Psychologists.length - 1].id) + 1).toString().
the first "(" is not necessary.

**Review by nicolasviola — CHANGES_REQUESTED** [review]

**Comment by ortizjoaquin** [comment]
corrections applied!

**Review by nicolasviola — APPROVED** [review]

**Pull request merged**